FireboltNETSDK/Client/FireboltParameter.cs

```diff
@@ -12,39 +13,53 @@ namespace FireboltDotNetSdk.Client
     /// </summary>
     public sealed class FireboltParameter : DbParameter
     {
+        internal const string defaultParameterName = "parameter";
         private static readonly Regex ParameterNameRegex = new Regex("^[a-zA-Z_][0-9a-zA-Z_]*$");
+        private static ISet<DbType> unsupportedTypes = new HashSet<DbType>()
+        {
+            DbType.Currency, DbType.VarNumeric, DbType.AnsiStringFixedLength, DbType.StringFixedLength, DbType.Xml, DbType.DateTime2, DbType.Single
+        };
+
+        private static IDictionary<Type, DbType> dbTypes = new Dictionary<Type, DbType>
+        {
+            { typeof(byte[]), DbType.Binary },
+            { typeof(byte), DbType.Byte },
+            { typeof(bool), DbType.Boolean },
+            { typeof(DateOnly), DbType.Date },
+            { typeof(DateTime), DbType.DateTime },
+            { typeof(float), DbType.Decimal },
+            { typeof(double), DbType.Double },
+            { typeof(decimal), DbType.Decimal },
+            { typeof(Guid), DbType.Guid },
+            { typeof(short), DbType.Int16 },
+            { typeof(int), DbType.Int32 },
+            { typeof(long), DbType.Int64 },
+            { typeof(sbyte), DbType.SByte },
+            { typeof(string), DbType.String },
+            { typeof(TimeOnly), DbType.Time },
+            { typeof(ushort), DbType.UInt16 },
+            { typeof(uint), DbType.UInt32 },
+            { typeof(ulong), DbType.UInt64 },
+            { typeof(DateTimeOffset), DbType.DateTimeOffset },
+        };
+
+        // This constant is copied from Microsoft's System.Data.SqlTypes.SQLDecimal
+        private const uint DBL_DIG = 17; // Max decimal digits of double
```

Comment:
Should this be Firebolt's value instead? We should be clear the precision is 15 and extra digits will be truncated.

Reply:
Fixed and added comment.